src/template.rs

```diff
@@ -29,20 +31,53 @@ pub struct Template {
     has_fields: bool,
 }
 
-impl From<&str> for Template {
-    fn from(src: &str) -> Template {
-        Template {
-            src: src.into(),
-            src_bytes: src.into(),
-            has_ts: StrftimeItems::new(src).filter(is_dynamic).count() > 0,
-            has_fields: RE.is_match(src),
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum TemplateError {
+    StrftimeError,
+}
+
+impl Error for TemplateError {}
+
+impl fmt::Display for TemplateError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::StrftimeError => write!(f, "Invalid strftime item"),
+        }
+    }
+}
+
+impl TryFrom<&str> for Template {
+    type Error = TemplateError;
+
+    fn try_from(src: &str) -> Result<Self, Self::Error> {
+        match StrftimeItems::new(src)
+            .filter(|item| matches!(item, Item::Error))
+            .count()
+        {
+            0 => Ok(Template {
+                src: src.into(),
+                src_bytes: src.into(),
+                has_ts: StrftimeItems::new(src).filter(is_dynamic).count() > 0,
+                has_fields: RE.is_match(src),
+            }),
+            _ => Err(TemplateError::StrftimeError),
```

Comment:
`StrftimeItems::new(src)` called twice, i.e. `src` also will be parsed twice.
Instead `.filter(|item| matches!(item, Item::Error)).count()` [.any](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) can be used.

Reply:
Yeah, that was not my finest piece of code, just wanted to get it written. I'll see how I can rework it.